### DISCUSSION TITLE
Change Editor

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#530] [Docs]
<meta path="kit/change-editor">
      
# Change Editor

This will re-prompt you to pick an editor from your PATH by updating your kenv `.env`.

You can always manually change the editor that Script Kit uses to open files in `~/.kenv/.env`.

The following would use `code` (assuming is on the "PATH").

```bash
KIT_EDITOR=code
```

If `code` isn't on your PATH, you can add the full path to the editor.

**Comment by ApolloTang** [comment]
In your answer

> This will re-prompt you to pick an editor from your PATH by updating your kenv .env

What does `This` in above refer to?

**Reply by johnlindquist** [reply]
The app loads in these GitHub discussions as a json file to display info for the options in the "Kit" and "Docs" tabs. This allows me to update the docs in the app based on feedback here.

The "Kit" tab -> "Change Editor" option will load in the first post in this thread you and I are discussing. So "this" refers to the "Change Editor" option in the app.

The wording is definitely confusing if you're not in the context of the app. I'll try to make App/Docs relationship more clear in the text.

**Reply by ApolloTang** [reply]
Thanks! that make sense now. I didn't know this GitHub discussions is linked from the Kit App.

**Comment by ApolloTang** [comment]
I am trying to change my editor to MacVim, 
I set the values in my `~/.kenv/.env` :

```
KIT_EDITOR=mvim
KIT_TERMINAL=iterm
```
where mvim is 

```
$ which mvim
/Applications/MacVim.app/Contents/bin/mvim
```

With the above, the Kit App stuck, and not responding. Any idea how to set my editor to MacVim ?

---
By the way I experimented with the following settings, they works, just can't get Kit to work with MacVim. 


```
KIT_EDITOR=vim
KIT_TERMINAL=iterm
```

where vim is:
```
$ ls -l $(which vim)
-rwxr-xr-x  1 <USER>  wheel  2994412 21 May  2020 /usr/local/bin/vim
```

The following also working
```
KIT_EDITOR=code
KIT_TERMINAL=iterm
```

Where code is

```
/usr/local/bin/code -> /Users/<USER>/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code
```

**Reply by johnlindquist** [reply]
Try setting "KIT_EDITOR" to the full path to MacVim.

(I'm on my phone now, but will attempt it tomorrow)

**Reply by ApolloTang** [reply]
Yes, it work with full path: 
```
KIT_EDITOR=/Applications/MacVim.app/Contents/bin/mvim
KIT_TERMINAL=iterm
```
Thank you! 
And thank you for the super quick response ⚡ ⚡ ⚡ ⚡
